overlord/stateengine.go

```diff
@@ -36,14 +36,20 @@ type StateManager interface {
 	Ensure() error
 }
 
-// StateWaiterStopper is implemented by StateManagers that have
-// running activities that can be waited or terminated.
-type StateWaiterStopper interface {
-	// Wait asks manager to wait for all running activities to finish.
+// StateWaiter is optionally implemented by StateManagers that have running
+// activities that can be waited.
+type StateWaiter interface {
+	// Wait asks manager to wait for all running activities to
+	// finish.
 	Wait()
+}
 
-	// Stop asks the manager to terminate all activities running concurrently.
-	// It must not return before these activities are finished.
+// StateStopper is optionally implemented by StateManagers that have
+// running activities that can be terminated.
+type StateStopper interface {
```

Comment:
Good idea with decoupling them, thanks for spotting the problem with ifacemgr/udevmon.